ingestion/src/metadata/ingestion/source/database/bigquery/helper.py

```diff
@@ -0,0 +1,66 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Source connection helper
+"""
+
+from typing import Any
+
+from pydantic import BaseModel
+from sqlalchemy import inspect
+
+from metadata.generated.schema.entity.services.connections.database.bigQueryConnection import (
+    BigQueryConnection,
+)
+from metadata.generated.schema.security.credentials.gcpValues import (
+    GcpCredentialsValues,
+    SingleProjectId,
+)
+from metadata.ingestion.source.connections import get_connection
+from metadata.utils.bigquery_utils import get_bigquery_client
+
+
+class InspectorWrapper(BaseModel):
+    client: Any
+    engine: Any
+    inspector: Any
+
+
+def get_inspector_details(
+    database_name: str, service_connection: BigQueryConnection
+) -> InspectorWrapper:
+    """
+    Method to get the bigquery inspector details
+    """
+    # TODO support location property in JSON Schema
+    # TODO support OAuth 2.0 scopes
```

Comment:
shall we create a couple of tickets for this?

Reply:
Umm sure we can, also not sure if this is being asked by the community, it was there in the code I just moved it here 😅

Reply:
ah ok, then @ayush-shah maybe you can chip in on what we are actually missing

Reply:
@ulixius9 @pmbrull location and oauth2 scopes to be added in bigqueryConnection if someone wants to modify those in order to connect to external resource
ref: https://github.com/open-metadata/OpenMetadata/pull/10127/files#r1100868327
cc: @yu-iskw